### PR TITLE
Add support for Microsoft Azure Blob Storage in Google Cloud Storage Transfer Service operators

### PR DESCRIPTION
**Pull request by khyurri**
Operators Code, examples and documentation completed. I've got bug on `CloudDataTransferServiceCancelOperationOperator` from example. I'll fix it.

Can you, please, review changed code?
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/cloud_storage_transfer_service.py [hunk 1]
```diff
@@ -97,8 +97,15 @@ class GcpTransferOperationStatus:
 TRANSFER_JOBS = 'transferJobs'
 TRANSFER_OPERATIONS = 'transferOperations'
 TRANSFER_OPTIONS = 'transfer_options'
-TRANSFER_SPEC = 'transferSpec'
+TRANSFER_SPEC: str = 'transferSpec'
```
Comment (ad-m) [inline airflow/providers/google/cloud/hooks/cloud_storage_transfer_service.py]: Why did you only specify the type here? Why is the type not added for the other constants?

Comment (khyurri) [reply airflow/providers/google/cloud/hooks/cloud_storage_transfer_service.py]: Seems it was automatic. I'll remove this type

**Comment by mik-laj** [comment]
@khyurri  Do you need any help? I am available for you if you have any questions.

**Comment by khyurri** [comment]
@mik-laj 
yes, I think I need help with previewing documentation. I cannot build documentation locally and cannot check it is correct

**Comment by ad-m** [comment]
@khyurri , how do you attempt to build documentation and what errors are you getting?

**Comment by potiuk** [comment]
Indeed we have a super-easy way to build documentation - simply run './breeze build-documentation' and all should happen automatically. I would love to hear what problems you experience with it.

**Comment by khyurri** [comment]
@ad-m
I build documentation using this command from repository root :

```shell
python setup.py build_sphinx
```
Build completes without any errors with message: `The HTML pages are in build/sphinx/html.`

I can find separate documentation pages (like in `operators/cloud_storage_transfer_service/index.html`), but find example usages (like `howto_operator_gcp_transfer_create_job_body_azure`).

**Comment by potiuk** [comment]
@khyurri - here is official way of build documentation (described in our CONTRIBUTING document):
https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#documentation

Additionally (if you do not want to setup your virtualenv, we have BREEZE development environment that downloads and uses Docker image to do the same: https://github.com/apache/airflow/blob/master/BREEZE.rst#using-the-airflow-breeze-environment

**Comment by khyurri** [comment]
@potiuk @ad-m 
I builded documentation locally, but now I can't find html for this file `airflow/providers/google/cloud/example_dags/example_cloud_storage_transfer_service_azure.py`

**Comment by mik-laj** [comment]
@khyurri Sorry. I missed your messsage.   Example DAG files are not available in the documentation until they are used in the documentation. You can attach a fragment of files using the exampleinclude directive. Documentation for this service can be found here.
https://github.com/apache/airflow/blob/master/docs/howto/operator/gcp/cloud_storage_transfer_service.rst

> Is it enough for me to check that the examples from the documentation work, or should I do something else?

I would be happy if you run example DAG and see if it works. You can do it with the help of automatic system tests.

I am very sad that you had to wait so long for my answer.

**Comment by khyurri** [comment]
I think it's done.

**Comment by turbaszek** [comment]
@khyurri can you rebase please? In the meantime we introduced black formatter for providers package. Sorry for such a long time of waiting for review.

CC: @michalslowikowski00 may be interested in this PR

**Comment by RosterIn** [comment]
@khyurri seems like rebase wasn't successful

**Comment by khyurri** [comment]
> seems like rebase wasn't successful

Yeah. Working on it.

**Comment by mik-laj** [comment]
@khyurri  Do you need help? I will gladly do it for you.

**Comment by khyurri** [comment]
@mik-laj 
Hey, sorry for delay. Could you please help me to rebase this PR?

**Comment by ad-m** [comment]
@khyurri  do you have experience in rebase?


/sparkle @mik-laj  for supporting Apache Airflow project!

**Comment by mik-laj** [comment]
> Hey, sorry for delay. Could you please help me to rebase this PR?

I was busy with Airflow 2.0, but today I will try to look at it.

**Comment by khyurri** [comment]
I rebased on master, black changed too much code.

File: airflow/providers/google/cloud/example_dags/example_cloud_storage_transfer_service_azure.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    # [START howto_operator_gcp_transfer_cancel_operation]
+    cancel_operation = CloudDataTransferServiceCancelOperationOperator(
+        task_id="cancel_operation",
+        operation_name="{{task_instance.xcom_pull("
+        "'wait_for_second_operation_to_start', key='sensed_operations')[0]['name']}}",
+    )
+    # [END howto_operator_gcp_transfer_cancel_operation]
+
+    # [START howto_operator_gcp_transfer_delete_job]
+    delete_transfer_from_aws_job = CloudDataTransferServiceDeleteJobOperator(
```
Comment (mik-laj) [inline airflow/providers/google/cloud/example_dags/example_cloud_storage_transfer_service_azure.py:190]:
```suggestion
    delete_transfer_from_job = CloudDataTransferServiceDeleteJobOperator(
```

File: airflow/providers/google/cloud/example_dags/example_cloud_storage_transfer_service_azure.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        task_id="delete_transfer_from_aws_job",
```
Comment (mik-laj) [inline airflow/providers/google/cloud/example_dags/example_cloud_storage_transfer_service_azure.py:191]:
```suggestion
        task_id="delete_transfer_from_job",
```

File: airflow/providers/google/cloud/example_dags/example_cloud_storage_transfer_service_azure.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        job_name="{{task_instance.xcom_pull('create_transfer_job_from_aws')['name']}}",
+        project_id=GCP_PROJECT_ID,
+    )
+    # [END howto_operator_gcp_transfer_delete_job]
+
+    create_transfer_job_from_azure >> wait_for_operation_to_start >> pause_operation
+    pause_operation >> list_operations >> get_operation >> resume_operation
+    resume_operation >> wait_for_operation_to_end >> cancel_operation
+    cancel_operation >> delete_transfer_from_aws_job
```
Comment (mik-laj) [inline airflow/providers/google/cloud/example_dags/example_cloud_storage_transfer_service_azure.py:200]:
```suggestion
    cancel_operation >> delete_transfer_from_job
```

**Comment by mik-laj** [comment]
Code looks good to me. Can you add tests to avoid regression?

**Comment by khyurri** [comment]
Sure

**Pull request closed without merging**